var/spack/repos/builtin/packages/dihydrogen/package.py

```diff
@@ -89,32 +77,29 @@ class Dihydrogen(CMakePackage, CudaPackage, ROCmPackage):
     for val in ROCmPackage.amdgpu_targets:
         depends_on('aluminum amdgpu_target=%s' % val, when='amdgpu_target=%s' % val)
 
-    depends_on('cuda', when=('+cuda' or '+legacy'))
-    depends_on('cudnn', when=('+cuda' or '+legacy'))
+    depends_on('cuda', when=('+cuda' or '+distconv'))
+    depends_on('cudnn', when=('+cuda' or '+distconv'))
```

Comment:
This won't work. Python will evaluate `'+cuda' or '+legacy'` to True since both strings are non-empty. What you want is:
```suggestion
    for when in ['+cuda', '+distconv']:
        depends_on('cuda', when=when)
        depends_on('cudnn', when=when)
```
Or something like that.